### PR TITLE
Add Command implementation in Developer guide

### LINKED ISSUE #210: Add command implementation to Developer Guide

### PR DESCRIPTION
**Pull request by keanecjy**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -10,6 +10,72 @@ title: Developer Guide
...
+**Alternative 1 (current choice)**: `AddCommand` is split into 4 other types of `AddItemCommand`. Parser parses the
+user input and creates the specific `AddItemCommand` for execution. The following activity diagram shows how the 
+execution of the `AddApplicationCommand` will work.
+
+![AddApplicationCommandActivityDiagram](images/AddApplicationCommandActivityDiagram.png)
+
+- Pros: 
+    - Each command has its own specific task to execute. This means that classes are more flexible and can be changed
+    very easily. 
+    - Higher cohesion as the class is only dependent on the one `Item` type
```
Comment (ZoroarkDarkrai) [inline docs/DeveloperGuide.md:110]: Why does it have higher cohesion?
Isn't low dependency --> low coupling?

**Review by ZoroarkDarkrai — COMMENTED** [review]
Nice work!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (keanecjy) [reply docs/DeveloperGuide.md:110]: Did you misread? 
Quoting the textbook: 
Cohesion is a measure of how strongly-related and focused the various responsibilities of a component are.

Comment (ZoroarkDarkrai) [reply docs/DeveloperGuide.md:110]: "Coupling is a measure of the degree of dependence between components, classes, methods, etc. Low coupling indicates that a component is less dependent on other components." ?

**Comment by keanecjy** [comment]
Generalize commands to use XCommand and YCommand instead

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (keanecjy) [reply docs/DeveloperGuide.md:110]: There's no coupling word used here?

Comment (ZoroarkDarkrai) [reply docs/DeveloperGuide.md:110]: "Only dependent on the one `Item` type" --> low dependency --> low coupling?

**Comment by ZoroarkDarkrai** [comment]
Nvm I'll just merge this first

**Review by ZoroarkDarkrai — APPROVED** [review]

**Pull request merged**